Added re pattern to exclude based on the basename of a file

stevenmburns opened the pull request:
This addresses issue #68

I added a command line argument `--pattern <re>`. If used, then source events on files with a base name that matches the regular expression specified by the argument with not trigger an update.

My primary use case is to avoid the false triggering caused by the lock file used by emacs.
The pattern to exclude these files is: `^\.#\S+$`
I added another command line argument (`-e` or `--emacs`) to do this directly.

xsteve commented:
I like the feature.

But I don't think that the name pattern is a good choice. I would never think that pattern specifies a list of files that should not be watched. A better name would be for example --dont-watch-files

Adding the --emacs option is not really necessary. It simplifies calling for emacs users (I am one of them - so I would probably use it). However I don't like the vision to add such a flag for other editors as well.

stevenmburns commented:
Good suggestions. I'll make the changes and tidy it up a bit (update the README.md)  later today.

stevenmburns commented:
Simplifed things a bit; removed the --emacs feature; the pattern is matching the whole basename now, instead of the basename with the extension stripped off.
Added a few lines to the README's as well.
The README.rst formatting changed a bit when I ran it through pandoc. I hope that is okay.

xsteve commented:
I would love to see this merged...
